app/controllers/api/database.php

```diff
@@ -2002,13 +1876,7 @@ function createAttribute(string $collectionId, Document $attribute, Response $re
     ->inject('usage')
     ->inject('events')
     ->inject('mode')
-    ->action(function ($collectionId, $documentId, $data, $read, $write, $response, $dbForProject, $audits, $usage, $events, $mode) {
-        /** @var Appwrite\Utopia\Response $response */
-        /** @var Utopia\Database\Database $dbForProject */
-        /** @var Appwrite\Event\Audit $audits */
-        /** @var Appwrite\Stats\Stats $usage */
-        /** @var Appwrite\Event\Event $events */
-        /** @var string $mode */
+    ->action(function (string $collectionId, string $documentId, $data, ?array $read, ?array $write, Response $response, Database $dbForProject, EventAudit $audits, Stats $usage, Event $events, string $mode) {
```

Comment:
Is there a reason we don't have a type for `$data`?
We could use `string|array $data` right ? Let me know what you think 👍

Reply:
I made the changes , I have given `$data` the type `array`

Reply:
Pretty sure it will never will be a string here 🤔

Reply:
Yeah, looks like there are some cases where it can be both, lets stick to `string|array $data`